### PR TITLE
cloud: add doc for sink to gcs

### PR DESCRIPTION
**Pull request by grovecai**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add sink to GCS support in TiDB Cloud

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-cloud/changefeed-sink-to-cloud-storage.md
Comment (ran-huang) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:1]: Preview: https://pingcap-docs-website-scaffold-git-preview-clou-e40220-ran-huang.vercel.app/tidbcloud/changefeed-sink-to-cloud-storage

File: tidb-cloud/changefeed-sink-to-cloud-storage.md [hunk 1]
```diff
@@ -18,36 +18,111 @@ This document describes how to create a changefeed to stream data from TiDB Clou
 - Because TiDB Cloud uses TiCDC to establish changefeeds, it has the same [restrictions as TiCDC](https://docs.pingcap.com/tidb/stable/ticdc-overview#unsupported-scenarios).
 - If the table to be replicated does not have a primary key or a non-null unique index, the absence of a unique constraint during replication could result in duplicated data being inserted downstream in some retry scenarios.
 
-## Create a changefeed
+## Step 1. Create a changefeed
```
Comment (ran-huang) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:21]: I broke the long process down into three major steps. They'll be easier to follow.

File: tidb-cloud/changefeed-sink-to-cloud-storage.md [hunk 2]
```diff
@@ -18,36 +18,111 @@ This document describes how to create a changefeed to stream data from TiDB Clou
...
-5. Customize **Table Filter** to filter the tables that you want to replicate. For the rule syntax, refer to [table filter rules](https://docs.pingcap.com/tidb/stable/ticdc-filter#changefeed-log-filters).
+![s3_endpoint](/media/tidb-cloud/changefeed/sink-to-cloud-storage-s3-endpoint.jpg)
+
+</div>
+<div label="GCS">
+
+For **GCS**, please fill the **GCS Endpoint** area.
+
+![gcs_endpoint](/media/tidb-cloud/changefeed/sink-to-cloud-storage-gcs-endpoint.jpg)
+
```
Comment (ran-huang) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:44]:
```suggestion
```

File: tidb-cloud/changefeed-sink-to-cloud-storage.md [hunk 3]
```diff
@@ -18,36 +18,111 @@ This document describes how to create a changefeed to stream data from TiDB Clou
...
+For **GCS**, before filling **GCS Endpoint**, you need to first grant the GCS bucket access and make the GCS bucket at the same region with your TiDB cluster. Take the following steps:
```
Comment (ran-huang) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:45]:
```suggestion
For **GCS**, before filling **GCS Endpoint**, you need to first grant the GCS bucket access. Take the following steps:
```

File: tidb-cloud/changefeed-sink-to-cloud-storage.md [hunk 4]
```diff
@@ -18,36 +18,111 @@ This document describes how to create a changefeed to stream data from TiDB Clou
...
+
+1. In the TiDB Cloud console, record the **Service Account ID** field's value, which will be used to grant TiDB Cloud access to your GCS bucket.
```
Comment (ran-huang) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:47]:
```suggestion
1. In the TiDB Cloud console, record the **Service Account ID**, which will be used later to grant TiDB Cloud access to your GCS bucket.
```

File: tidb-cloud/changefeed-sink-to-cloud-storage.md [hunk 5]
```diff
@@ -18,36 +18,111 @@ This document describes how to create a changefeed to stream data from TiDB Clou
...
+        - storage.objects.delete
+
+        ![Add permissions](/media/tidb-cloud/changefeed/sink-to-cloud-storage-gcs-assign-permission.png)
+
+3. Go to the [Bucket](https://console.cloud.google.com/storage/browser) page, and click the name of the GCS bucket you want TiDB Cloud to access.
+4. On the **Bucket details** page, click the **Permissions** tab, and then click **Grant access**.
+
+    ![Grant Access to the bucket ](/media/tidb-cloud/changefeed/sink-to-cloud-storage-gcs-grant-access-1.png)
+
+5. Fill in the following information to grant access to your bucket, and then click **SAVE**.
```
Comment (ran-huang) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:76]:
```suggestion
5. Fill in the following information to grant access to your bucket, and then click **Save**.
```

File: tidb-cloud/changefeed-sink-to-cloud-storage.md [hunk 6]
```diff
@@ -18,36 +18,111 @@ This document describes how to create a changefeed to stream data from TiDB Clou
...
+
+    ![Get bucket URI](/media/tidb-cloud/changefeed/sink-to-cloud-storage-gcs-uri01.png)
+    
+    For example, if the folder name is `test-sink-gcs`, then the URI would be `gs://test-sink-gcs/`.
+
+    If you want to use a folder's gsutil URI, open the folder, and then click the copy button add `gs://` as prefix.. 
+
+    ![Get bucket URI](/media/tidb-cloud/changefeed/sink-to-cloud-storage-gcs-uri02.png)
+    
+    For example, if the folder name is `changefeed-xxx`, then the URI would be `gs://test-sink-gcs/changefeed-xxx` as the URI.
```
Comment (ran-huang) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:97]: 下一步里只需要用到 bucket URI，这里是不是没必要讲 folder gsutil URI？

Comment (grovecai) [reply tidb-cloud/changefeed-sink-to-cloud-storage.md]: 应该是下一步的名词用得不好， 两种都可能使用

File: tidb-cloud/changefeed-sink-to-cloud-storage.md [hunk 7]
```diff
@@ -18,43 +18,108 @@ This document describes how to create a changefeed to stream data from TiDB Clou
 - Because TiDB Cloud uses TiCDC to establish changefeeds, it has the same [restrictions as TiCDC](https://docs.pingcap.com/tidb/stable/ticdc-overview#unsupported-scenarios).
 - If the table to be replicated does not have a primary key or a non-null unique index, the absence of a unique constraint during replication could result in duplicated data being inserted downstream in some retry scenarios.
 
-## Create a changefeed
+## Step 1. Configure destination
 
-1. Navigate to the cluster overview page of the target TiDB cluster, and then click **Changefeed** in the left navigation pane.
+Navigate to the cluster overview page of the target TiDB cluster. Click **Changefeed** in the left navigation pane,click **Create Changefeed**, and select **Amazon S3** or **GCS** as the destination. The configuration process varies depend on the destination you choose.
```
Comment (ran-huang) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:23]:
```suggestion
Navigate to the cluster overview page of the target TiDB cluster. Click **Changefeed** in the left navigation pane, click **Create Changefeed**, and select **Amazon S3** or **GCS** as the destination. The configuration process varies depend on the destination you choose.
```

Comment (ran-huang) [reply tidb-cloud/changefeed-sink-to-cloud-storage.md:23]: 这里是只能选当前集群的 provider 吗？GCP 的集群没有看到 S3 的 destination 选项

File: tidb-cloud/changefeed-sink-to-cloud-storage.md [hunk 8]
```diff
@@ -65,27 +130,38 @@ This document describes how to create a changefeed to stream data from TiDB Clou
...
 
     </div>
     </SimpleTab>
 
-8. Click **Next** to configure your changefeed specification.
+4. In the **Flush Parameters** area, you can configure two items:
+
+   ![Flush Parameters](/media/tidb-cloud/changefeed/sink-to-cloud-storage-flush-parameters.jpg)
+    - **Flush Interval**, with a default setting of 60 seconds, adjustable within a range of 2 seconds to 10 minutes;
+    - **File Size**, with a default setting of 64 MB, adjustable within a range of 1 MB to 512 MB.
```
Comment (ran-huang) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:147]:
```suggestion

    - **Flush Interval**: set to 60 seconds by default, adjustable within a range of 2 seconds to 10 minutes;
    - **File Size**: set to 64 MB by default, adjustable within a range of 1 MB to 512 MB.

```

File: tidb-cloud/changefeed-sink-to-cloud-storage.md [hunk 9]
```diff
@@ -65,27 +130,38 @@ This document describes how to create a changefeed to stream data from TiDB Clou
...
+   > **Note:**
+   > These two parameters will affect the quantity of objects generated in cloud storage, and they apply to each individual database table. If there are a large number of database tables, the same configuration will result in a corresponding increase in the number of objects generated, leading to an associated rise in the cost of invoking the cloud storage API. Therefore, it is advisable to configure the parameters appropriately based on your RPO (Recovery Point Objective) and cost requirements.
```
Comment (ran-huang) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:149]:
```suggestion
    > **Note:**
    >
    > These two parameters will affect the quantity of objects generated in cloud storage for each individual database table. If there are a large number of tables, using the same configuration will increase the number of objects generated and subsequently raise the cost of invoking the cloud storage API. 
    >
    > Therefore, it is recommended to configure these parameters appropriately based on your Recovery Point Objective (RPO) and cost requirements.
```

File: tidb-cloud/changefeed-sink-to-cloud-storage.md (hunk 7 above)
Comment (grovecai) [reply tidb-cloud/changefeed-sink-to-cloud-storage.md:23]: 是的

File: tidb-cloud/changefeed-sink-to-cloud-storage.md [hunk 10]
```diff
@@ -18,36 +18,101 @@ This document describes how to create a changefeed to stream data from TiDB Clou
...
+
+    ![Get bucket URI](/media/tidb-cloud/changefeed/sink-to-cloud-storage-gcs-uri01.png)
+
+    For example, if the folder name is `test-sink-gcs`, then the URI would be `gs://test-sink-gcs/`.
+
+    If you want to use a folder's gsutil URI, open the folder, and then click the copy button add `gs://` as prefix..
+
+    ![Get bucket URI](/media/tidb-cloud/changefeed/sink-to-cloud-storage-gcs-uri02.png)
+
+    For example, if the folder name is `changefeed-xxx`, then the URI would be `gs://test-sink-gcs/changefeed-xxx` as the URI.
```
Comment (ran-huang) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:87]:
```suggestion
    - To get a bucket's gsutil URI, click the copy button and add `gs://` as a prefix. For example, if the folder name is `test-sink-gcs`, the URI would be `gs://test-sink-gcs/`.

        ![Get bucket URI](/media/tidb-cloud/changefeed/sink-to-cloud-storage-gcs-uri01.png)

    - To get a folder's gsutil URI, open the folder, and then click the copy button add `gs://` as a prefix. For example, if the folder name is `changefeed-xxx`, the URI would be `gs://test-sink-gcs/changefeed-xxx`.

        ![Get bucket URI](/media/tidb-cloud/changefeed/sink-to-cloud-storage-gcs-uri02.png)
```

File: tidb-cloud/changefeed-sink-to-cloud-storage.md [hunk 11]
```diff
@@ -18,36 +18,101 @@ This document describes how to create a changefeed to stream data from TiDB Clou
...
+
+7. In the TiDB Cloud console, go to the Changefeed's "Configure Destination" page, paste the GCS bucket gsutil URI to the **Bucket gsutil** URI field.
```
Comment (ran-huang) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:89]:
```suggestion
7. In the TiDB Cloud console, go to the Changefeed's **Configure Destination** page, and fill in the **bucket gsutil URI** field.
```

File: tidb-cloud/changefeed-sink-to-cloud-storage.md [hunk 12]
```diff
@@ -75,17 +140,33 @@ This document describes how to create a changefeed to stream data from TiDB Clou
     </div>
     </SimpleTab>
 
-8. Click **Next** to configure your changefeed specification.
+4. In the **Flush Parameters** area, you can configure two items:
+
+    ![Flush Parameters](/media/tidb-cloud/changefeed/sink-to-cloud-storage-flush-parameters.jpg)
+
+    - **Flush Interval**: set to 60 seconds by default, adjustable within a range of 2 seconds to 10 minutes;
+    - **File Size**: set to 64 MB by default, adjustable within a range of 1 MB to 512 MB.
```
Comment (ran-huang) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:148]:
```suggestion
    - **Flush Interval**: set to 60 seconds by default, adjustable within a range of 2 seconds to 10 minutes;
    - **File Size**: set to 64 MB by default, adjustable within a range of 1 MB to 512 MB.

    ![Flush Parameters](/media/tidb-cloud/changefeed/sink-to-cloud-storage-flush-parameters.jpg)
```

File: tidb-cloud/changefeed-sink-to-cloud-storage.md [hunk 13]
```diff
@@ -75,17 +140,33 @@ This document describes how to create a changefeed to stream data from TiDB Clou
...
+
+    > **Note:**
+    >
+    > These two parameters will affect the quantity of objects generated in cloud storage for each individual database table. If there are a large number of tables, using the same configuration will increase the number of objects generated and subsequently raise the cost of invoking the cloud storage API. 
+    >
+    > Therefore, it is recommended to configure these parameters appropriately based on your Recovery Point Objective (RPO) and cost requirements.
```
Comment (ran-huang) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:154]:
```suggestion
    > These two parameters will affect the quantity of objects generated in cloud storage for each individual database table. If there are a large number of tables, using the same configuration will increase the number of objects generated and subsequently raise the cost of invoking the cloud storage API. Therefore, it is recommended to configure these parameters appropriately based on your Recovery Point Objective (RPO) and cost requirements.
```

File: tidb-cloud/changefeed-sink-to-cloud-storage.md [hunk 14]
```diff
@@ -18,36 +18,97 @@ This document describes how to create a changefeed to stream data from TiDB Clou
...
+    - In the **New Principals** field, paste the **Service Account ID** of the target TiDB cluster you recorded before.
+    - In the **Select a role** drop-down list, type the name of the IAM role you just created, and then choose the name from the filter result.
+
+    > **Note:**
+    >
+    > To remove the access to TiDB Cloud, simply remove the access that you have granted.
+
+6. On the **Bucket details** page, click the **Objects** tab.
+
+    - To get a bucket's gsutil URI, click the copy button and add `gs://` as a prefix. For example, if the folder name is `test-sink-gcs`, the URI would be `gs://test-sink-gcs/`.
```
Comment (grovecai) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:77]: "folder name" => "bucket name"

File: tidb-cloud/changefeed-sink-to-cloud-storage.md [hunk 15]
```diff
@@ -18,36 +18,97 @@ This document describes how to create a changefeed to stream data from TiDB Clou
...
+
+        ![Get bucket URI](/media/tidb-cloud/changefeed/sink-to-cloud-storage-gcs-uri01.png)
+
+    - To get a folder's gsutil URI, open the folder, and then click the copy button add `gs://` as a prefix. For example, if the folder name is `changefeed-xxx`, the URI would be `gs://test-sink-gcs/changefeed-xxx`.
```
Comment (grovecai) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:81]: "if the folder name is `changefeed-xxx`" => if the bucket name is `test-sink-gcs` and the folder name is `changefeed-xxx`

File: tidb-cloud/changefeed-sink-to-cloud-storage.md [hunk 16]
```diff
@@ -1,11 +1,11 @@
 ---
 title: Sink to Cloud Storage
-Summary: Learn how to create a changefeed to stream data from a TiDB Dedicated cluster to cloud storage, such as Amazon S3.
+Summary: Learn how to create a changefeed to stream data from a TiDB Dedicated cluster to cloud storage, such as Amazon S3, GCS.
```
Comment (hfxsd) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:3]:
```suggestion
Summary: Learn how to create a changefeed to stream data from a TiDB Dedicated cluster to cloud storage, such as Amazon S3 and GCS.
```

File: tidb-cloud/changefeed-sink-to-cloud-storage.md [hunk 17]
```diff
@@ -1,11 +1,11 @@
...
 ---
 
 # Sink to Cloud Storage
 
-This document describes how to create a changefeed to stream data from TiDB Cloud to cloud storage. Currently, only Amazon S3 is supported.
+This document describes how to create a changefeed to stream data from TiDB Cloud to cloud storage. Currently, only Amazon S3 and GCS are supported.
```
Comment (hfxsd) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:8]:
```suggestion
This document describes how to create a changefeed to stream data from TiDB Cloud to cloud storage. Currently, Amazon S3 and GCS are supported.
```

File: tidb-cloud/changefeed-sink-to-cloud-storage.md
Comment (grovecai) [reply tidb-cloud/changefeed-sink-to-cloud-storage.md:1]: 这个图片在 preview 里显示不出来， 知道为什么？
Screen Shot 2023-08-17 at 14 41 22

File: tidb-cloud/changefeed-sink-to-cloud-storage.md [hunk 18]
```diff
@@ -18,36 +18,97 @@ This document describes how to create a changefeed to stream data from TiDB Clou
...
-    - If yes, you are directed to the next step of configuration.
-    - If not, a connectivity error is displayed, and you need to handle the error. After the error is resolved, click **Next** to retry the connection.
+For **GCS**, before filling **GCS Endpoint**, you need to first grant the GCS bucket access. Take the following steps:
 
-5. Customize **Table Filter** to filter the tables that you want to replicate. For the rule syntax, refer to [table filter rules](https://docs.pingcap.com/tidb/stable/ticdc-filter#changefeed-log-filters).
+1. In the TiDB Cloud console, record the **Service Account ID**, which will be used to grant TiDB Cloud access to your GCS bucket.
+
+    ![gcs_endpoint](/media/tidb-cloud/changefeed/sink-to-cloud-storage-gcs-endpoint.png)
+
+2. In the Google Cloud Platform (GCP) Management Console, create an IAM role for your GCS bucket.
```
Comment (hfxsd) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:41]: GCP 已经改名叫 Google Cloud https://cloud.google.com/blog/topics/developers-practitioners/introducing-new-homepage-google-cloud

```suggestion
2. In the Google Cloud Management Console, create an IAM role for your GCS bucket.
```

File: tidb-cloud/changefeed-sink-to-cloud-storage.md [hunk 19]
```diff
@@ -18,36 +18,97 @@ This document describes how to create a changefeed to stream data from TiDB Clou
...
+
+    1. Sign in to the [GCP Management Console](https://console.cloud.google.com/).
```
Comment (hfxsd) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:43]:
```suggestion
    1. Sign in to the [Google Cloud Management Console](https://console.cloud.google.com/).
```

File: tidb-cloud/changefeed-sink-to-cloud-storage.md (hunk 15 above)
Comment (hfxsd) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md]:
```suggestion
    - To get a folder's gsutil URI, open the folder, and then click the copy button and add `gs://` as a prefix. For example, if the folder name is `changefeed-xxx`, the URI would be `gs://test-sink-gcs/changefeed-xxx`.
```

File: tidb-cloud/changefeed-sink-to-cloud-storage.md
Comment (ran-huang) [reply tidb-cloud/changefeed-sink-to-cloud-storage.md:1]: 因为不是这个 PR 里新增的图片，现在的 preview 方式拿不到。可以忽略

File: tidb-cloud/changefeed-sink-to-cloud-storage.md [hunk 20]
```diff
@@ -18,36 +18,97 @@ This document describes how to create a changefeed to stream data from TiDB Clou
...
+    >
+    > To remove the access to TiDB Cloud, simply remove the access that you have granted.
+
+6. On the **Bucket details** page, click the **Objects** tab.
+
+    - To get a bucket's gsutil URI, click the copy button and add `gs://` as a prefix. For example, if the bucket name is `test-sink-gcs`, the URI would be `gs://test-sink-gcs/`.
+
+        ![Get bucket URI](/media/tidb-cloud/changefeed/sink-to-cloud-storage-gcs-uri01.png)
+
+    - To get a folder's gsutil URI, open the folder, and then click the copy button add `gs://` as a prefix. For example, if the bucket name is `test-sink-gcs` and the folder name is `changefeed-xxx`, the URI would be `gs://test-sink-gcs/changefeed-xxx`.
```
Comment (ran-huang) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:81]:
```suggestion
    - To get a folder's gsutil URI, open the folder, click the copy button, and add `gs://` as a prefix. For example, if the bucket name is `test-sink-gcs` and the folder name is `changefeed-xxx`, the URI would be `gs://test-sink-gcs/changefeed-xxx`.
```

**Review by cchenkey — APPROVED** [review]

**Review by ran-huang — APPROVED** [review]

File: tidb-cloud/changefeed-sink-to-cloud-storage.md [hunk 21]
```diff
@@ -18,36 +18,97 @@ This document describes how to create a changefeed to stream data from TiDB Clou
...
+
+    3. Enter a name, description, ID, and role launch stage for the role. The role name cannot be changed after the role is created. 
+    4. Click **Add permissions**. Add the following read-only permissions to the role, and then click **Add**.
+
+        - storage.buckets.get
+        - storage.objects.create
+        - storage.objects.update
+        - storage.objects.get
+        - storage.objects.list
+        - storage.objects.delete
```
Comment (hfxsd) [inline tidb-cloud/changefeed-sink-to-cloud-storage.md:56]:
```suggestion
        - storage.buckets.get
        - storage.objects.create
        - storage.objects.delete        
        - storage.objects.get
        - storage.objects.list
        - storage.objects.update        
```

**Review by hfxsd — APPROVED** [review]

**Pull request merged**